cmake/BuildFlags.cmake

```diff
@@ -52,7 +52,12 @@ target_compile_options(cse_common_interface INTERFACE
     /nologo       # Suppresses display of sign-on banner.
     /fp:precise   #+Specifies floating-point behavior.
     /fp:except-   # Specifies floating-point behavior.
-    /arch:IA32    # Specifies the architecture for code generation (no special instructions).
+    $<$<STREQUAL:"${CSEARCHBUILDFLAG}","Win32">:
+      /arch:IA32    # Specifies the architecture for code generation (no special instructions).
+    >
+    $<$<STREQUAL:"${CSEARCHBUILDFLAG}","x64">:
+      /arch:AVX    # Enables the use of Intel Advanced Vector Extensions instructions.
```

Comment:
@nealkruis Is this an ok flag to replace IA32 when using x64? https://learn.microsoft.com/en-us/cpp/build/reference/arch-x64?view=msvc-170

Another option is to leave it empty which uses the flag SSE2. Let me know what you think about it.